test/test_tracer.cpp

```diff
@@ -2037,6 +2042,50 @@ TEST_TRACER("APM tracing disabled") {
   }
 }
 
+#if defined(__linux__) || defined(__unix__)
+TEST_TRACER("process discovery") {
+  auto find_memfd = []() -> std::optional<std::string> {
+    DIR* dir = opendir("/proc/self/fd");
+    if (dir == nullptr) return nullopt;
+
+    char path[PATH_MAX];
+    char target[PATH_MAX];
+
+    for (struct dirent* entry = readdir(dir); entry != nullptr;
+         entry = readdir(dir)) {
+      if (entry->d_type != DT_LNK) continue;
+
+      snprintf(path, sizeof(path), "/proc/self/fd/%s", entry->d_name);
+      auto len = readlink(path, target, sizeof(target) - 1);
+      if (len == -1) continue;
+      target[len] = '\0';
+
+      if (starts_with(target, "/memfd:datadog-tracer-info-")) {
+        closedir(dir);
+        return path;
+      }
+    }
+
+    return nullopt;
```

Comment:
Do you also need to call `closedir(dir)` here?

Reply:
Indeed. Good catch @zacharycmontoya 
```suggestion
    closedir(dir);
    return nullopt;
```